### ISSUE TITLE
Crash on shakespeare sample

### ISSUE THREAD

**Issue by leopd** [karpathy/char-rnn#8]
On brand-new AWS GPU machine (g2.2xlarge running ami-6238470a) I install the code and run it with 

th train.lua -data_dir data/tinyshakespeare/

It dies with

using CUDA on GPU 0...	
loading data files...	
cutting off end of data so that the batches/sequences divide evenly	
reshaping tensor...	
data load done. Number of batches in train: 211, val: 11, test: 1	
vocab size: 65	
creating an LSTM with 2 layers	
number of parameters in the model: 154165	
cloning criterion	
cloning softmax	
cloning embed	
cloning rnn	
/home/ubuntu/torch-distro/install/bin/luajit: ./util/OneHot.lua:17: attempt to call method 'long' (a nil value)
stack traceback:
	./util/OneHot.lua:17: in function 'forward'
	train.lua:172: in function 'opfunc'
	...ntu/torch-distro/install/share/lua/5.1/optim/rmsprop.lua:32: in function 'rmsprop'
	train.lua:226: in main chunk
	[C]: in function 'dofile'
	...rch-distro/install/lib/luarocks/rocks/trepl/scm-1/bin/th:131: in main chunk
	[C]: at 0x00406170

**Comment by karpathy** [comment]
Hello, have a look here: https://github.com/karpathy/char-rnn/issues/2

**Comment by leopd** [comment]
THanks.  Also, Just switching to ami-c79b7eac, which uses CUDA 7 (vs 6.5) fixes the problem.
